### PR TITLE
test(http-logger): fetch request and response body test case

### PR DESCRIPTION
**Pull request by shuaijinchao**
### What this PR does / why we need it:
FIX https://the-asf.slack.com/archives/CUC5MN17A/p1646136677114399

### Pre-submission checklist:
* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: t/lib/server.lua [hunk 1]
```diff
@@ -46,6 +46,14 @@ local function inject_headers()
     end
 end
 
+local function str_split(str, reps)
```
Comment (spacewander) [inline t/lib/server.lua:49]: Let's add the mock API in the test file like:
https://github.com/apache/apisix/blob/4feab8e6bf1cf8ffd16440848b693ce14a792fbc/t/plugin/clickhouse-logger.t#L39

It's not commonly used.

Comment (shuaijinchao) [reply t/lib/server.lua]: done

File: t/plugin/http-logger2.t [hunk 2]
```diff
@@ -32,6 +32,64 @@ add_block_preprocessor(sub {
...
+                local query = args.query or nil
+                ngx.req.read_body()
+                local body = ngx.req.get_body_data()
+
+                if query then
+                    if type(query) == "string" then
+                        query = {query}
+                    end
+
+                    local data, err = require("cjson").decode(body)
```
Comment (starsz) [inline t/plugin/http-logger2.t:65]: Hi, Why do we need to decode the body?

**Review by tzssangglass — APPROVED** [review]

File: t/plugin/http-logger2.t (hunk 2 above)
Comment (shuaijinchao) [reply t/plugin/http-logger2.t:65]: The log format data is JSON, and we need to get the information in the structure.

File: t/plugin/http-logger2.t [hunk 3]
```diff
@@ -32,6 +32,64 @@ add_block_preprocessor(sub {
...
+                        local val
+                        for j = 1, #fields do
+                            local key = fields[j]
+                            if j == 1 then
+                                val = data[key]
+                            else
+                                val = val[key]
+                            end
+                        end
+                        ngx.log(ngx.ERR ,query[i], ":", val)
```
Comment (spacewander) [inline t/plugin/http-logger2.t:81]: Better to use WARN log level

File: t/plugin/http-logger2.t [hunk 4]
```diff
@@ -141,3 +199,86 @@ GET /opentracing
...
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:12001": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/http-logger/test"
+                }]],
+                [[{
```
Comment (spacewander) [inline t/plugin/http-logger2.t:234]: We should remove the response body check

File: t/plugin/http-logger2.t (hunk 2 above)
Comment (starsz) [reply t/plugin/http-logger2.t:65]: Got it.

**Review by starsz — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**